gax/src/gax.ts

```diff
@@ -292,23 +326,133 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+export function checkRetryOptions(
```

Comment:
The name of the function is misleading. If it's `check...`, it should probably return a boolean. But it actually modifies the given `options` and returns the updated object, so the name should start from `convert...`, `update...`, or something like that.